### DISCUSSION TITLE
Procedure to build multi image with mcuboot on Zephyr for i.MX RT1160 EVK

### DISCUSSION THREAD

**Discussion by vishnusudhan** [zephyrproject-rtos/zephyr#45787] [General]
As i am new to this project , I am using i.MX RT 1160 EVK ,

where we can refer the detailed steps for multi-image build & flash with MCUBoot on zephyr for i.MX RT 1160 EVK? If its available share the link? I am also looking for the configuration , when we try to upgrade image over various method? like swap with out swap.

Also i have seen option in MCUBoot config like as below,

![Screenshot from 2022-05-19 11-14-36](https://user-images.githubusercontent.com/104553024/169218887-23b20375-3eff-4c31-a14f-7456e6b97d2a.png)

If i set this option as  (number of upgrades) 2 , based on my understanding this option use to enable to upgrade the multi image with MCUBoot. is that correct? I am getting error as like below when i enabling this option . kindly clarify , how to config when we use to update the multi image with MCUBoot over WEST/JTAG or USB/UART. 

![Screenshot from 2022-05-19 11-21-25](https://user-images.githubusercontent.com/104553024/169219711-1a90477c-f510-4313-811e-9b199d70baae.png)

cc:  @[danie007](https://github.com/danie007) & @tkmozhi

**Comment by DerekSnell** [comment]
Hi @vishnusudhan ,
I have not tried using MCUboot on the RT1160.  But I do have steps posted at [MCUboot and Zephyr on NXP i.MX RT series](https://github.com/zephyrproject-rtos/zephyr/discussions/45252), which uses a similar RT MCU.  I do not have experience with MCUboot's number of upgrades configuration.  But the steps at this link show how to have two application images stored in Slot0 and Slot1, how to download a new image to Slot1, and how to swap them.

Best regards
